echo_server.rb

```diff
@@ -0,0 +1,47 @@
+class UserEcho
+
+	def initialize(welcome, instructions)
+		@welcome = "Welcome!"
+		@instructions = "Please enter a word to be reprinted, to exit this program please enter exit:"
+	end
+
+	def welcome_message
+		@welcome
+	end
+
+	def instruction_message
+		@instructions
+	end
+
+	def start
+		@welcome
+		@instructions
+		input
+	end
+
```

Comment:
Think about your public facing interface.

Perhaps the below methods could be private?